tests/scripts/arm64_post_build.py

```diff
@@ -0,0 +1,302 @@
+g_test_url = "https://clrjit.blob.core.windows.net/arm64ci/CoreCLRTests-675622bb85f3c78de1967a78052d7280a2834611.zip"
+g_x64_client_url = "https://clrjit.blob.core.windows.net/arm64ci/x64_client.zip"
+
+################################################################################
+# Argument Parser
+################################################################################
+
+description = """Python script to facilitate running an arm64/arm test run using
+                 the cloud.
+              """
+
+parser = argparse.ArgumentParser(description=description)
+
+parser.add_argument("--force_update", dest="force_update", action="store_true", default=False)
+
+parser.add_argument("-repo_root", dest="repo_root", nargs='?', default=None)
+parser.add_argument("-arch", dest="arch", nargs='?', default=None)
+parser.add_argument("-build_type", dest="build_type", nargs='?', default=None)
+parser.add_argument("-scenario", dest="scenario", nargs='?', default=None)
+parser.add_argument("-key_location", dest="key_location", nargs='?', default=None)
+
+################################################################################
+# Helper Functions
+################################################################################
+
+def add_item_to_path(location):
+   """ Add the dotnet install to the path
+   """
+
+   os.environ["PATH"] = location + ";" + os.environ["PATH"]
+
+def copy_core_root(core_root):
+   """ Copy the core root directory to the current dir as "build"
+   Args:
+      core_root (str): location of the core_root directory
+   Returns:
+      copy_location (str): name of the location, for now hardcoded to build
+                         : for backcompat in the old system
+   """
+
+   new_location = "build"
+
+   # Delete used instances.
+   if os.path.isdir(new_location):
+      shutil.rmtree(new_location)
+
+   try:
+      shutil.copytree(core_root, new_location)
+
+   except OSError as error:
+      log("Core Root not copied. Error: %s" % error)
+
+   return new_location
+
+def log(message):
+   """ Helper function to print logging information
+   Args:
+      message (str): message to be printed
+   """
+
+   print "[arm64_post_build]: %s" % (message)
+
+def setup_cli(force_update=False):
+   """ Install the dotnet cli onto the machine
+   Args:
+      force_update (bool): whether or not to force an update. 
+   Return:
+      install_location (str): location of the installed cli
+   Notes:
+   This will be installed to %USERPROFILE%\dotnet. If force update is False
+   then we will not install the cli if it already exists.
+   
+   """
+   global g_dotnet_url
+
+   install_path = os.path.join(os.environ["USERPROFILE"], "dotnet")
+
+   # Only install if the cli doesn't exist or we are forcing an update
+   if not os.path.isdir(install_path) or force_update:
+      log("Downloading the .NET CLI")
+      if os.path.isdir(install_path):
+         shutil.rmtree(install_path)
```

Comment:
Possible race condition? I think the way I've done it in the past is to do a try/catch where you try to remove the tree, and if there's an exception, you assert that the path is not a dir.